### PR TITLE
add profile attribute

### LINKED ISSUE #99: criticality flag
When Endorser creates reference measurements, the generalized verifier needs to know the criticality of the reference values. There are several ways to communicate criticality.

1) The Attester Evidence contains criticality. If Verifier doesn't recognize the Evidence then Verifier must fail.
- difficult to ensure all forms of Evidence supports this (EAT, TcBInfo, SPDM, etc...)

2) Endorsements contains criticality. If the Verifier doesn't recognize the Reference Values, then fail.
2.a) Add criticality flag inside the triple - difficult to do since we've frozen.
2.b) Add criticality as an extension to the tag.
2.c) Add criticality to a profile.
2.d) Add a supplemental RIM that contains criticality information.

2a is probably the cleanest but hardest to do at this point.

### PR DESCRIPTION
**Pull request by thomas-fossati**
add an optional profile attribute to the unsigned-corim-map

Fixes #99

### CODE REVIEW

**Review by nedmsmith — COMMENTED** [review]
I believe it should be one-or-more<profile-type-choice> since the semantics are that the verifier must recognize the profile or fail (and it is reasonable that verifiers can recognize more than one profile).

**Comment by thomas-fossati** [comment]
> I believe it should be one-or-more since the semantics are that the verifier must recognize the profile or fail (and it is reasonable that verifiers can recognize more than one profile).

Interesting observation.

File: unsigned-corim.cddl [hunk 1]
```diff
@@ -2,9 +2,12 @@ unsigned-corim-map = {
   corim.id => $corim-id-type-choice
   corim.tags => one-or-more<$concise-tag-type-choice>
   ? corim.dependent-rims => one-or-more<corim-locator-map>
+  ? corim.profile => one-or-more<profile-type-choice>
```
Comment (yogeshbdeshpande) [inline unsigned-corim.cddl:5]: Do we really want profile to be a "Closed Type Choice" ?
I think, we can leave the flexibility in the interface to let it be extensible for a possible use expansion for future?

Comment (nedmsmith) [reply unsigned-corim.cddl:5]: I originally thought it should be $class-Id-type-choice. But we have vendor in class-map that resolves the namespace authority for uuids. Uri uses domain name to resolve ns Auth. And you can define a Uri that includes a uuid. So I don’t think it needs to be an extensible type choice.

**Pull request merged**